Plugin.pm

```diff
@@ -628,6 +635,164 @@ sub getFeatured {
 	});
 }
 
+sub getHome {
+	my ( $client, $cb, $args, $params ) = @_;
+
+	getAPIHandler($client)->page(sub {
+		my $modules = shift;
+
+		my $items = [ map {
+			{
+				name => $_->{title},
+				type => 'link',
+				url => $_->{type} eq 'HIGHLIGHT_MODULE' ? \&getHighlights : \&getModule,
+				passthrough => [ { module => $_ } ],
+			}
+		} @$modules ];
+
+		$cb->( {
+			items => $items
+		} );
+	}, 'pages/home' );
+}
+
+sub getHighlights {
+	my ( $client, $cb, $args, $params ) = @_;
+
+	my $module = $params->{module};
+	my $items = [];
+
+	foreach my $entry (@{$module->{highlights}}) {
+		my $item;
+		my $title = $entry->{title};
+
+		$entry = $entry->{item};
+
+		# formats are not exactly consistent with the rest of the API. Small changes
+		# but requires a few exceptions. We could change getImageUrl and typeOfItem...
+		if ($entry->{type} eq 'ALBUM') {
+			$item = _renderAlbum($entry->{item}, { addArtistToTitle => 1 });
+		}
+		elsif ($entry->{type} eq 'PLAYLIST') {
+			$item = _renderPlaylist($entry->{item});
+			# image can't be 1280x1280...
+			my $image = $entry->{item}->{squareImage} =~ s/-/\//gr;
+			$item->{image} = IURL . $image . "/640x640.jpg";
+		}
+		elsif ($entry->{type} =~ /MIX/) {
+			$item = _renderMix($client, $entry->{item});
+			# image uses a different key (S/M/L) than current MIX
+			$item->{image} = $entry->{item}->{images}->{MEDIUM}->{url} || $entry->{item}->{images}->{SMALL}->{url} || $entry->{item}->{images}->{LARGE}->{url};;
```

Comment:
If that image selection was implemented in `getImageUrl` you wouldn't even have to call it here: `_renderMix()` would do it for you.